src/compiletest.md

```diff
@@ -0,0 +1,183 @@
+# `compiletest`
+## Introduction
+`compiletest` is the main test harness of the Rust test suite.  It allows test authors to organize large numbers of tests (the
+Rust compiler has many thousands), efficient test execution (parallel execution is supported), and allows the test author to
+configure behavior and expected results of both individual and groups of tests.
+
+`compiletest` tests may check test code for success, for failure or in some cases, even failure to compile.  Tests are
+typically organized as a Rust source file with annotations in comments before and/or within the test code, which serve to
+direct `compiletest` on if or how to run the test, what behavior to expect, and more.
+
+The tests themselves are typically (but not always) organized into "suites"--for example, `run-pass`, a folder
+representing tests that should succeed, `run-fail`, a folder holding tests that should compile successfully, but return
+a failure (non-zero status), `compile-fail`, a folder holding tests that should fail to compile, and many more.  The various
+suites are defined in `src/tools/compiletest/src/common.rs` in the `pub struct Config` declaration.  And a very good 
+introduction to the different suites of compiler tests along with details about them can be found
+[here, at Brian Anderson's blog](https://brson.github.io/2017/07/10/how-rust-is-tested#s-ct).
```

Comment:
I think we should also link to the chapter that will be added in #47

Reply:
Do you want a link in the doc to a PR or to wait until it lands and can be linked to under `master`?

Reply:
> to wait until it lands and can be linked to under master

I think this would be best, unless you are under a time constraint.